ethereumj-core/src/main/java/org/ethereum/sharding/processing/consensus/BeaconStateTransition.java

```diff
@@ -67,8 +84,19 @@ public BeaconState applyBlock(Beacon block, BeaconState to) {
                     .withDynasty(dynasty)
                     .withLastStateRecalc(cycleStartSlot(block))
                     .withFinality(finality);
+
+            // remove attestations older than last_state_recalc
+            List<AttestationRecord> uptodateAttestations = new ArrayList<>();
```

Comment:
I am starting to think of decoupling crystallized and active state transitions when looking at these lines.